### ISSUE TITLE
Publishing NUnit(2) xml via Publishing Task does not return values in ITmiTestImplementation

### ISSUE THREAD

**Issue by Tyonoro** [Microsoft/azure-pipelines-tasks#9042]
## Troubleshooting
We run our test via NUnit 2 and 3 Testrunner. Publishing the nunit.xml works fine if you want to see the test results in TFS but there is no way to get the results via ITmiTestImplementation. 
Our workaround at the moment is to transform the XML into a TRX, fill the needed information and publish by VSTest/TRX setting of publisher. In this way we receive our values in ITmiTestImplementation.
Is this behaviour correct/intended? Is there a better way instead of doing the transformation?
The presence of 
* full qualified test case name
* its namespace/assembly
* and an unique id (reproducible stable)
is crucial for our project!

## Environment
- Microsoft Visual Studio Team Foundation Server (On-Premise)
Version 15.117.27024.0: 
- Agent - Hosted:  
  -Win10 x64

## Issue Description
CommandLine Tasks
Publish test result task
NUnit 2 and 3 Framework/Runner

**Comment by divais** [comment]
Publishing nunit.xml helps you publish data in TFS for reporting needs. You can use our APIs to get the required data. https://docs.microsoft.com/en-us/rest/api/azure/devops/test/results/get?view=azure-devops-rest-5.0

**Comment by GraueEminenz79** [comment]
If I get @Tyonoro right, he states that -e.g. given the example of your link-
```
"automatedTestStorage": "unittestproject1.dll",
"automatedTestType": "UnitTest",
"automatedTestTypeId": "13cdc9d9-ddb5-4fa4-a97d-d965ccfc6d4b",
"automatedTestId": "aefba017-ab06-be36-6b92-de4e29836f72"
```
is ___not present/filled___ in TFS _although_ available (assembly) or ascertainable (testId) from NUnit results when using Test Result Publishing Task.

**Comment by kapilratra** [comment]
@Tyonoro . We were trying to understand why are you using ITmiTestImplementation .
 Are you using Client OM and doing some operation on ITmiTestImplementation  once you fetch results.

You can start consuming our REST Apis instead of SOAP client OM . Will anyways like to understand the usage

**Comment by domischell** [comment]
@kapilratra . let me answer that for @Tyonoro: we want to see long-term trends of test cases e.g. flakiness. But due to TFS' retention policy the builds/test case results are thrown away after a while. Therefore, we copy the test cases and their results into a database and use the Guid as primary key.

In the XAML world this worked fine, but not in vNext's Publisher Task and that's why we started this thread. In the REST-APIs' replies we found the data from which we can generate the Guid ourselves, so from our side the issue is solved.

Thanks for your time!
